docs/sources/tempo/configuration/_index.md

```diff
@@ -1231,6 +1235,226 @@ memberlist:
+  [cas_retry_delay: <duration> | default 1s]
+
+etcd:
+  [endpoints: <list of string> | default = [] ]
+  [dial_timeout: <duration> | default = 10s]
+  [max_retries: <int> | default = 10 ]
+  [tls_enabled: <bool> | default = false]
+
+  # TLS config
+  [tls_cert_path: <string> | default = ""]
+  [tls_key_path: <string> | default = ""]
+  [tls_ca_path: <string> | default = ""]
+  [tls_server_name: <string> | default = ""]
+  [tls_insecure_skip_verify: <bool> | default = false]
+  [tls_cipher_suites: <string> | default = ""]
+  [tls_min_version: <string> | default = ""]
+
+  [username: <string> | default = ""]
+  [password: <secret string> | default = ""]
+
+multi:
+  [primary: <string> | default = ""]
+  [secondary: <string> | default = ""]
+  [mirror_enabled: <bool> | default = false]
+  [mirror_timeout: <bool> | default = 2s]
+```
+
+### Search config
+
+```yaml
+# Target number of bytes per GET request while scanning blocks. Default is 1MB. Reducing
+# this value could positively impact trace search performance at the cost of more requests
+# to object storage.
+[chunk_size_bytes: <uint32> | default = 1000000]
+
+# Number of traces to prefetch while scanning blocks. Default is 1000. Increasing this value
+# can improve trace search performance at the cost of memory.
+[prefetch_trace_count: <int> | default = 1000]
+
+# Number of read buffers used when performing search on a vparquet block. This value times the  read_buffer_size_bytes
+# is the total amount of bytes used for buffering when performing search on a parquet block.
+[read_buffer_count: <int> | default = 32]
+
+# Size of read buffers used when performing search on a vparquet block. This value times the read_buffer_count
+# is the total amount of bytes used for buffering when performing search on a parquet block.
+[read_buffer_size_bytes: <int> | default = 1048576]
+
+# Granular cache control settings for parquet metadata objects
+# Deprecated. See [Cache](#cache) section.
+cache_control:
+
+    # Specifies if footer should be cached
+    [footer: <bool> | default = false]
+
+    # Specifies if column index should be cached
+    [column_index: <bool> | default = false]
+
+    # Specifies if offset index should be cached
+    [offset_index: <bool> | default = false]
+```
+
+### WAL config
+
+The storage WAL configuration block.
+
+```yaml
+# Where to store the wal files while they are being appended to.
+# Must be set.
+# Example: "/var/tempo/wal
+[path: <string> | default = ""]
+
+# Where to store the completed wal files
+# If not set (""), will join the `path` with "completed" to generate the effective path
+# Example: "/var/tempo/wal/completed"
+[completedfilepath: <string> | default = join(.path, "/completed")]
+
+# Where to store the intermediate blocks while they are being appended to.
+# Will always join the `path` with "blocks" to generate the effective path
+# Example: "/var/tempo/wal/blocks" (ignored)
+[blocksfilepath: <ignored> | = join(.path, "/blocks")]
```

Comment:
I think I found a bug?

I described what the code does here, but I don't think this is the intended behaviour...

Please verify :

https://github.com/grafana/tempo/blob/3a4f309b0d9413fc8762daaa0bd9769a08cc8bf8/tempodb/wal/wal.go#L70-L75


I think it's supposed to look like this (but with `c.BlocksFilepath == ""`):
https://github.com/grafana/tempo/blob/3a4f309b0d9413fc8762daaa0bd9769a08cc8bf8/tempodb/wal/wal.go#L56-L67

Reply:
You are right @WesselAtWork it is overwriting the config always. Please feel free to create a PR.

Reply:
@WesselAtWork , I've created a bug issue here: https://github.com/grafana/tempo/issues/3902

Would you add additional details?